compiler-cli/src/run.rs

```diff
@@ -34,6 +34,20 @@ pub fn command(
                 module: mod_path.to_owned(),
             });
         }
+
+        if mod_path.starts_with("src/") {
+            return Err(Error::InvalidModulePrefix {
+                module: mod_path.strip_prefix("src/").unwrap().to_string(),
```

Comment:
unwrapping seemed safe here because we do a starts_with check. but what's a neater way of handling this? 

i considered handling the option directly or use unwrap_or_default, but both didnt feel right

example 1:
```
 match mod_path.strip_prefix("src/") {
                Some(path) => {
                    return Err(Error::InvalidModulePrefix {
                        module: path.to_string(),
                        prefix: "src/".to_string(),
                    });
                }
                _ => (),
            }
```

Reply:
We'll still need to check if the module exists as `src/src/wibble.gleam` is a valid module.

So we'll need to look up the module and if it does not exist and it starts with src or test, then check if it exists once this prefix has been removed.

Reply:
src/src/x is valid? Okay I’ll def check. 

1. Do I simply check that a file exists by using the std fs or is there something else you have in mind? I had thought you’d built a dependency graph prior so I can just check that, but direct file check was my initial thought.
2. Are you able to run a module within a third party package? Double checking here.

Reply:
> unwrapping seemed safe here because we do a starts_with check. but what's a neater way of handling this?

You can use if let for this!.

```rust
if let Some(path) = mod_path.strip_prefix("src/") {
	// logic
}
```

Reply:
> Do I simply check that a file exists by using the std fs or is there something else you have in mind? I had thought you’d built a dependency graph prior so I can just check that, but direct file check was my initial thought.

You can use this function here, to match how we do it for the correct module name.

https://github.com/gleam-lang/gleam/blob/deb9f26fd66b33bcc5e2d7e96f0707d1519419b9/compiler-cli/src/run.rs#L72

Any checks for `src/` and `test/` should go after this existing check for the module name as-is.

> Are you able to run a module within a third party package? Double checking here.

Yup! This will all be handled by that method.

Reply:
Thanks yall. I believe it's working correctly now. I wrote my manual testing strategy in the description. 
I feel inclined to write at least a couple tests here, but is this advised? And how should I go about it?
I updated the shell file `test/running_modules/run_tests.sh` as an initial test, but I'm not asserting anything yet.

Reply:
We don't have a good way to test this sort of build tool stuff yet I'm afraid :(